slapp/transfers/upload.py

```diff
@@ -93,30 +99,38 @@ def run(self, db_conn: query_utils.DbConnection):
                 return_index=True)
         experiment_ids = [manifests[ui]['experiment-id'] for ui in uindex]
         self.logger.info(f"{full_video_paths.size} full videos to upload")
-        s3_full_videos = {}
+
+        args = []
         for eid, video_path in zip(experiment_ids, full_video_paths):
             object_key = prefix + "/" + f"{eid}_"
             object_key += pathlib.PurePath(video_path).name
-            s3_full_video = utils.upload_file(
-                    video_path,
-                    self.args['s3_bucket_name'],
-                    object_key)
-            s3_full_videos[video_path] = s3_full_video
+            args.append((
+                video_path,
+                self.args['s3_bucket_name'],
+                object_key))
+
+        # NOTE a reason to use ThreadPool instead of Pool is that
+        # moto testing does not work with a process-based pool
```

Comment:
oh, interesting

Reply:
I think it is a [known issue](https://github.com/spulec/moto/issues/2111).

It does not throw an error, the files expected in the bucket aren't there for anything that went through `multiprocessing.Pool`